core/src/main/java/com/linecorp/armeria/server/healthcheck/HealthCheckService.java

```diff
@@ -136,6 +138,8 @@ public static HealthCheckServiceBuilder builder() {
     private final long maxLongPollingTimeoutMillis;
     private final double longPollingTimeoutJitterRate;
     private final long pingIntervalMillis;
+    private final ReentrantLock lock = new ReentrantLock();
+    @GuardedBy("lock")
```

Comment:
How about removing this annotation because the lock doesn't guard `healthCheckerListener`?